src/octree/types.rs

```diff
@@ -31,46 +31,59 @@ pub(in crate::octree) enum NodeChildrenArray<T: Default> {
 #[cfg_attr(feature = "serialization", derive(Serialize, Deserialize))]
 pub(in crate::octree) struct NodeChildren<T: Default> {
     /// The key value to signify "no child" at a given slot
-    pub(in crate::octree) default_key: T,
+    pub(in crate::octree) no_child_marker_key: T,
 
     /// The contained child key values
     pub(in crate::octree) content: NodeChildrenArray<T>,
 }
 
 pub trait VoxelData {
-    fn new(r: u8, g: u8, b: u8, a: u8, user_data: u32) -> Self;
+    fn new(color: Albedo, user_data: u32) -> Self;
     /// The color to display during raytracing 0-255 RGBA
-    fn albedo(&self) -> [u8; 4];
+    fn albedo(&self) -> Albedo;
     /// User defined data
     fn user_data(&self) -> u32;
-    /// determines if the voxel is to be hit by rays in the raytracing algorithms
+    /// Determines if the voxel is to be hit by rays in the raytracing algorithms
     fn is_empty(&self) -> bool {
-        [0, 0, 0, 0] == self.albedo() && 0 == self.user_data()
+        self.albedo().is_transparent() && self.user_data() == 0
```

Comment:
Is it enough to go by `Transparency` or should it really just be `0,0,0,0` so, basically `Transparent Black`?

Reply:
The best would be to leave this up to the implementer, but for ease the default behavior should be whatever generically feels logical. 

The reasoning behind `Transparent Black´ is that all Bytes are unset.
The reasoning behind `Transparent` is that whatever it may be doesn't matter because it can not be seen anyway. 

I think Both are good, because if somebody needs to store something, but not display it, they can do so with user data. 

I think going with `self.albedo().is_transparent() && self.user_data() == 0` would do the trick